### PR TITLE
Pass the dataSource object directly in chipRenderer

### PR DESCRIPTION
**Pull request by oktay-sen**
Having the chip object itself as input to chipRenderer will allow adding properties like avatarUrl or backgroundColor to the chip object, and rendering the chips based on these properties.

### CODE REVIEW

File: src/ChipInput.js [hunk 1]
```diff
@@ -508,6 +508,7 @@ class ChipInput extends React.Component {
               return chipRenderer({
                 value,
                 text: dataSourceConfig ? tag[dataSourceConfig.text] : tag,
+                chip:tag,
```
Comment (leMaik) [inline src/ChipInput.js]: Please just add a space here (e.g. lint the code), then it's fine. :+1: 
Also, please update the README accordingly.

Comment (oktay-sen) [inline src/ChipInput.js]: Done!

File: README.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ import ChipInput from 'material-ui-chip-input'
...
+| chipRenderer | `function` | | A function of the type `({ value, text, chip, isFocused, isDisabled, handleClick, handleRequestDelete, defaultStyle }, key) => node` that returns a chip based on the given properties. This can be used to customize chip styles.  Each item in the `dataSource` array will be passed to `chipRenderer` as arguments `value` and `text`. If `dataSource` is an array of objects and `dataSourceConfig` is present, then `value` and `text` will instead correspond to the object values defined in `dataSourceConfig`. If `dataSourceConfig` is not set and `dataSource` is an array of objects, then a custom `chipRenderer` must be set.|
```
Comment (leMaik) [inline README.md]: You should also explain what `chip` is. (Sorry for being so nick-picky…)

Comment (oktay-sen) [inline README.md]: No problem, I changed it.

> Each item in the `dataSource` array will be passed to `chipRenderer` as arguments `chip`, `value` and `text`.

I think it describes the functionality accurately, but might not be verbose enough.

**Comment by oktay-sen** [comment]
Cheers! Sorry for asking but when can I expect to see it in npm?

**Comment by leMaik** [comment]
Soon :tm: I'll publish it later today (maybe tomorrow).

**Comment by leMaik** [comment]
@oktay-sen Just released v0.16.0, which adds this feature!

**Comment by oktay-sen** [comment]
That's great news! Thank you :)

On Aug 2, 2017 20:30, "Maik Marschner" <notifications@github.com> wrote:

@oktay-sen <https://github.com/oktay-sen> Just released v0.16.0, which adds
this feature!

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/TeamWertarbyte/material-ui-chip-input/pull/130#issuecomment-319774042>,
or mute the thread

**Pull request merged**